libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.StreamingExtensions;
+using Microsoft.Bot.StreamingExtensions.Transport;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// Used to process incoming requests sent over an <see cref="IStreamingTransport"/> and adhering to the Bot Framework Protocol v3 with Streaming Extensions.
+    /// </summary>
+    public class StreamingRequestHandler : RequestHandler
+    {
+#if DEBUG
+        public
+#else
+        private
+#endif
+        string userAgent;
+
+        private IBot bot;
+
+        private IServiceProvider services;
+
```

Comment:
Readonly. Also, it's weird how it's only set in one constructor, but not the other.